src/Dotenv.cs

```diff
@@ -55,7 +62,18 @@ public static Dotenv Load(Stream input)
         /// <returns>Dictionary of environment variables.</returns>
         public Dictionary<string, string> GetVariables()
         {
-            return variables;
+            // Return a copy so caller cannot modify internal state.
+            return CloneVariables( );
+        }
+
+        private Dictionary<string, string> CloneVariables()
```

Comment:
Yes I mean `CleanVariables`, it should have a comment. All methods and properties in the source files should have a code comment. Test files don't.